addons/evm/src/codec/crypto.rs

```diff
@@ -0,0 +1,43 @@
+use alloy::signers::k256::ecdsa::SigningKey;
+use alloy_signer_local::{coins_bip39::English, LocalSigner, MnemonicBuilder};
+use hmac::digest::generic_array::GenericArray;
+
+use crate::signers::DEFAULT_DERIVATION_PATH;
+
+pub type UnsafeSigner = LocalSigner<SigningKey>;
+pub fn mnemonic_to_unsafe_signer(
+    mnemonic: &str,
+    derivation_path: Option<&str>,
+    is_encrypted: Option<bool>,
+    password: Option<&str>,
+) -> Result<UnsafeSigner, String> {
+    if is_encrypted.is_some() {
+        return Err("encrypted mnemonic signers are not yet supported".to_string());
+    }
+    let derivation_path = derivation_path.unwrap_or(DEFAULT_DERIVATION_PATH);
+
+    let mut mnemonic_builder = MnemonicBuilder::<English>::default()
+        .phrase(mnemonic)
+        .derivation_path(derivation_path)
+        .map_err(|e| format!("failed to instantiate unsafe signer from mnemonic: {e}"))?;
```

Comment:
If developers are using `secret_key` for construct key these signers, I think we should stick to the `secret_key` designation for helping debugging their failing constructs.